builder/bundles.go

```diff
@@ -342,8 +342,16 @@ func queryRpmFullPath(packageCmd []string, pkgName string, repo string) (string,
 	if pURL.Scheme == "file" { // obtain the full path without the url scheme
 		pURL.Scheme = ""
 		rpmFullPath = pURL.String()
+		// If repo cache dir has not been guessed before, guess it just one time.
+		// This is an attempt to improve performance in case the rpms are not stored directly within the repo baseurl.
+		if !repos[repo].guessCache {
+			repoInfo := repos[repo]
+			repoInfo.guessCache = true
+			repoInfo.cacheDir = filepath.Dir(rpmFullPath)
+			repos[repo] = repoInfo
```

Comment:
Why is `repoInfo` made? Couldn't we directly update repos[repo] with less code?

```
repos[repo].guessCache = true
repos[repo].cacheDir = filepath.Dir(rpmFullPath)
```

Reply:
go doesn't let you do that assignment. Its a compile error. I could maybe pass the repo map as a pointer.